src/sdk/env.rs

```diff
@@ -0,0 +1,88 @@
+//! EnvResourceDetector
+//!
+//! Implementation of ResourceDetector in extract information from environment.
+use crate::sdk::resource::ResourceDetector;
+use crate::sdk::Resource;
+use crate::api::{KeyValue, Key, Value};
+use std::env;
+use std::time::Duration;
+
+const OTEL_RESOURCE_ATTRIBUTES: &str = "OTEL_RESOURCE_ATTRIBUTES";
+
+/// Resource detector implements ResourceDetector and is used to extractor general SDK configuration from environment
+/// See https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#telemetry-sdk for semantic conventions.
+#[derive(Debug)]
+pub struct EnvResourceDetector {}
```

Comment:
we've been using `_private` here so we could add more without breaking changes (forces people to use the `new` method) e.g. https://github.com/open-telemetry/opentelemetry-rust/blob/e0bc3648110b38d338dfc84180b085185c1b517a/src/api/correlation/propagation.rs#L17-L19

Reply:
Good point!